src/agent/adu_core_interface/tests/adu_core_interface_ut.cpp

```diff
@@ -76,7 +77,12 @@ class ADUC_UT_ReportPropertyAsyncValues
     {
         deviceHandle = deviceHandleIn;
 
+        const size_t maxLen = sizeof(reportedState)/sizeof(reportedState[0]);
+        if (reportedStateLenIn >= maxLen) {
+            throw std::invalid_argument("reportedStateLenIn exceeded max length!");
+        }
         memcpy(reportedState, reportedStateIn, reportedStateLenIn);
+        reportedState[reportedStateLenIn] = '\0';
```

Comment:
Since we're using memcpy here, I'm assuming reportedStateLenIn is size of data, in bytes.

Perhaps, maxlen should be sizeof(reportedState) * sizeof(reportedState[0]), instead of array size.

Or... (if reportedStateLenIn is character count)

memcpy(reportedState, reportedStateIn, (reportedStateLenIn + 1) * sizeof(*reportedState))

Our tests expect string data type, so, we could also use strncpy instead of memcpy...

And dynamically allocate reportedState, so that we don't have to throw an exception.

Reply:
Good feedback.  I changed it to use std::string, which also allowed removing several reinterpret_cast.